Elias-Hackyourtemperature ii

EliasRajha opened the pull request.

dyaboykyl commented on assignments/hackyourtemperature/app.js:
```diff
@@ -0,0 +1,30 @@
+    if (response.ok) {
+      const temperature = data.main.temp;
+      return res.send({ weatherText: `The current temperature in ${cityName} is ${temperature}°C` });
+    } else {
+      return res.status(404).send({ weatherText: "City is not found!" });
```
I like that you're handling all situations in which `response.ok === false` but you might not necessarily want to assume that a 404 is always the right return code. For example, you could receive a 401 from the weather service if your api key is invalid or expired, even if the city name was valid. That would result in the client getting a 404, which would be confusing.

I won't block the review on that, but it's something to think about - what might be a better way to handle other errors besides 404 that the weather service could return?

dyaboykyl commented on assignments/hackyourtemperature/app.js:
```diff
@@ -0,0 +1,30 @@
+import express from 'express';
+import fetch from 'node-fetch';
```
When I run your tests locally I get an error on this line. Are they working for you? Please double check that your jest.config.js and babel.config.cjs files are correct

EliasRajha commented:
-I changed my jest.config.js file so it can handle ES modules better with Babel.
-I added a plugin to babel.config.cjs to help with module imports.
-I also installed version 2 of node-fetch to avoid problems.
-I found a mistake in the error message for invalid cities in my test file (app.test.js), so I fixed it to match the actual message ("City not found!").
